src/main/drivers/rx/rx_sx1280.c

```diff
@@ -0,0 +1,500 @@
+    sx1280WriteCommand(SX1280_RADIO_SET_AUTOFS, 0x01);                        //enable auto FS
+    sx1280WriteRegister(0x0891, (sx1280ReadRegister(0x0891) | 0xC0));         //default is low power mode, switch to high sensitivity instead
+    sx1280SetPacketParams(12, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, SX1280_LORA_IQ_NORMAL); //default params
+    sx1280SetFrequencyHZ(2400000000);                                         //Step 3: Set Freq
+    sx1280SetFIFOaddr(0x00, 0x00);                                            //Step 4: Config FIFO addr
+    sx1280SetDioIrqParams(SX1280_IRQ_RADIO_ALL, SX1280_IRQ_TX_DONE | SX1280_IRQ_RX_DONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE); //set IRQ to both RXdone/TXdone on DIO1
+}
+
+void sx1280Config(const sx1280_lora_bandwidths_e bw, const sx1280_lora_spreading_factors_e sf, const sx1280_lora_coding_rates_e cr, const uint32_t freq, const uint8_t preambleLength, const bool iqInverted)
+{
+    sx1280SetMode(SX1280_MODE_SLEEP);
+    sx1280PollBusy();
+
+    sx1280ConfigLoraDefaults();
+    sx1280SetOutputPower(13); //default is max power (12.5dBm for SX1280 RX)
+    sx1280SetMode(SX1280_MODE_STDBY_XOSC); 
+    sx1280ClearIrqStatus(SX1280_IRQ_RADIO_ALL);
+    sx1280ConfigLoraModParams(bw, sf, cr);
+    sx1280SetPacketParams(preambleLength, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, (sx1280_lora_iq_modes_e)((uint8_t)!iqInverted << 6)); // TODO don't make static etc.
+    sx1280SetFrequencyReg(freq);
+}
+
+void sx1280SetOutputPower(const int8_t power)
+{
+    uint8_t buf[2];
+    buf[0] = power + 18;
+    buf[1] = (uint8_t) SX1280_RADIO_RAMP_04_US;
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_TXPARAMS, buf, 2);
+}
+
+void sx1280SetPacketParams(const uint8_t preambleLength, const sx1280_lora_packet_lengths_modes_e headerType, const uint8_t payloadLength, const sx1280_lora_crc_modes_e crc, const sx1280_lora_iq_modes_e invertIQ)
+{
+    uint8_t buf[7];
+
+    buf[0] = preambleLength;
+    buf[1] = headerType;
+    buf[2] = payloadLength;
+    buf[3] = crc;
+    buf[4] = invertIQ;
+    buf[5] = 0x00;
+    buf[6] = 0x00;
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_PACKETPARAMS, buf, 7);
+}
+
+void sx1280SetMode(const sx1280_operating_modes_e opMode)
+{
+    uint8_t buf[3];
+
+    switch (opMode)
+    {
+
+    case SX1280_MODE_SLEEP:
+        sx1280WriteCommand(SX1280_RADIO_SET_SLEEP, 0x01);
+        break;
+
+    case SX1280_MODE_CALIBRATION:
+        break;
+
+    case SX1280_MODE_STDBY_RC:
+        sx1280WriteCommand(SX1280_RADIO_SET_STANDBY, SX1280_STDBY_RC);
+        break;
+        
+    case SX1280_MODE_STDBY_XOSC:
+        sx1280WriteCommand(SX1280_RADIO_SET_STANDBY, SX1280_STDBY_XOSC);
+        break;
+
+    case SX1280_MODE_FS:
+        sx1280WriteCommand(SX1280_RADIO_SET_FS, 0x00);
+        break;
+
+    case SX1280_MODE_RX:
+        buf[0] = 0x00; // periodBase = 1ms, page 71 datasheet, set to FF for cont RX
```

Comment:
also can we have this without all the empty lines between cases?

Reply:
removed empty lines